cmd/public-ip/main.go

```diff
@@ -10,14 +11,27 @@ import (
 )
 
 func main() {
+	// Url to get IP info
+	var urlForIp string = "https://ifconfig.co/ip"
+	var urlForCity string = "https://ifconfig.co/city"
 
 	// Set display texts to defaults.
 	var fullText string = "unknown"
 	var shortText string = "unknown"
+	var whatIsMyIpUrl string
+
+	var cityFlag = flag.Bool("city", false, "pass -city to report City instead of IP")
+	flag.Parse()
+
+	if cityFlag != nil && *cityFlag {
```

Comment:
```suggestion
	if *cityFlag {
```

Reply:
As you set a default value for the ```-city``` flag (```false```), the case that the flag would evaluate to ```nil``` cannot exist. The second check is sufficient.